src/plugins/initiatives/createGraph.js

```diff
@@ -35,6 +37,12 @@ function initiativeNode(initiative: Initiative): Node {
   };
 }
 
+export function initiativeWeight(initiative: Initiative): ?NodeWeight {
+  if (!initiative.weight) return;
+  const {incomplete, complete} = initiative.weight;
+  return !initiative.completed ? incomplete : complete;
+}
+
```

Comment:
Ok, I can see the case for letting the initiative.weight be null (in which case default); in that case, shouldn't this function contain the "default-setting" behavior?

I think the default weights should be `{incomplete: 0, complete: 1}`, which we could easily implement in this function.

Reply:
Imo, default setting does not belong (at all) in this function. This is simply a conversion from `?InitiativeWeight` to `?NodeWeight`.

The result of `null` from an algorithm point of view should be to fall back on the node type weight. Which you can achieve in one of two ways:
A) With a noop multiplication `{incomplete: 1, complete: 1}`.
B) Not setting a weight at the individual node address at all.
This type simply allows you to choose both approaches, and prefers B as it's less expensive.

The value you're suggesting should be a configuration option that we don't currently support. What you really want is for the PluginDeclaration to set `{incomplete: 0, complete: 1}` as the default and allow overriding it at the node type level. I say that because "we don't give cred to incomplete initiatives" is a sane default, but definitely debatable. Other communities might want `{incomplete: 0.25, complete: 8}`.

For now, the _workaround_ for not having this flexibility is to require all the JSON files have weights.

Reply:
As I feel this discussion is more a configuration discussion than a technical issue with this PR, going to merge as-is :slightly_smiling_face: